### PR TITLE
docs: document breaking change for nvim_create_autocmd callback

### PR DESCRIPTION
**Pull request by gpanders**
https://github.com/neovim/neovim/pull/27428 changed the semantics of callbacks passed to nvim_buf_attach and nvim_create_autocmd such that any truthy value will detach the callback/delete the autocommand (rather than just the literal boolean value `true`). Update the documentation accordingly and add an entry to `news.txt`.

cc @bfredl

### CODE REVIEW

**Comment by bfredl** [comment]
it should be changed back (and tested)

**Comment by bfredl** [comment]
after a bit more discussion, the past behavior of `nvim_create_autocmd` was actually inconsistent with `nvim_buf_attach` and similar functions. perhaps we should rip of the bandaid and accept a breaking change to make it consistent (but still document it, of course).

**Comment by gpanders** [comment]
I was mistaken that this is a breaking change for `nvim_buf_attach`: this is in fact how those callbacks have always worked. It is only `nvim_create_autocmd()` that has had its behavior changed.

The fact that `nvim_create_autocmd` behaved differently from `nvim_buf_attach` in the first place was not intentional so can be considered a bug. Making the two consistent with each other is a good fix. I've updated the news entry accordingly.

File: src/nvim/api/buffer.c [hunk 1]
```diff
@@ -125,7 +125,7 @@ Integer nvim_buf_line_count(Buffer buffer, Error *err)
 ///             - on_bytes: Lua callback invoked on change.
 ///               This callback receives more granular information about the
 ///               change compared to on_lines.
-///               Return `true` to detach.
+///               Return a truthy value (not `false` or `nil`) to detach.
```
Comment (zeertzjq) [inline src/nvim/api/buffer.c:128]: There is another mention at `on_lines`

Comment (gpanders) [reply src/nvim/api/buffer.c]: 73c87c5aead2c21eec0889f9d86952a0bee4f8aa

**Pull request merged**